prow/github/client.go

```diff
@@ -177,29 +177,39 @@ func (c *Client) Throttle(hourlyTokens, burst int) {
 }
 
 // NewClient creates a new fully operational GitHub client.
-func NewClient(token, base string) *Client {
+// 'token' is the GitHub access token to use.
+// 'bases' is a ';' delimited array of endpoints to use in order of preference.
+//   An endpoint is used when all preceding endpoints have returned a conn err.
+//   This should be used when using the ghproxy GitHub proxy cache to allow
+//   this client to bypass the cache if it is temporarily unavailable.
+func NewClient(token string, bases []string) *Client {
```

Comment:
This might be easier to use in a few places if bases were variadic. Some of the code would not change. https://github.com/kubernetes/test-infra/pull/8050/files#r188838729